cardinal/plugins.py

```diff
@@ -485,16 +571,17 @@ def unload(self, plugins):
                 failed_plugins.append(plugin)
                 continue
 
-                try:
-                    self._close_plugin_instance(plugin)
-                except Exception:
-                    # Log the exception that came from trying to unload the
-                    # plugin, but don't skip over the plugin. We'll still
-                    # unload it.
-                    self.logger.exception(
-                        "Didn't close plugin cleanly: %s" % plugin
-                    )
-                    failed_plugins.append(plugin)
+            try:
+                self._unregister_plugin_events(plugin)
```

Comment:
I don't think this can throw an exception, so we can probably move it above, outside of the `try` statement. I'd prefer not to have it in here, because if it ever does start throwing exceptions, I'd still like to try to call `self._close_plugin_instance()` *after* we fail to unregister our callbacks. There's some other cleanup that goes on in there that we can still attempt. So if it ever does throw an exception in the future, it'll probably be two separate `try/except` blocks.